modules/redis/src/main/scala/zio/redis/api/Subscription.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis.api
+
+import zio.redis.ResultBuilder.ResultStreamBuilder1
+import zio.redis._
+import zio.redis.api.Subscription.PubSubCallback
+import zio.redis.internal._
+import zio.schema.Schema
+import zio.stream.Stream
+import zio.{Chunk, IO, UIO}
+
+trait Subscription extends SubscribeEnvironment {
+
+  final def subscribe(channel: String): ResultStreamBuilder1[Id] =
+    subscribeWithCallback(channel)(None, None)
+
+  final def subscribeWithCallback(channel: String)(
+    onSubscribe: Option[PubSubCallback],
+    onUnsubscribe: Option[PubSubCallback]
```

Comment:
yeah it seems to make more ergonomic api. I changed to use `NoopCallback`. 
I changed function name `subscribe` to `subscribeSingle` for only single channel subscription call because it's more clear to distinct multiple channels subscription call (also they have different return type signature)
[`5ca98e7` (#714)](https://github.com/zio/zio-redis/pull/714/commits/5ca98e7d763e1b85998623aeabf4935f8fe69405)